lookup_plugins/oo_option.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+# vim: expandtab:tabstop=4:shiftwidth=4
+
+'''
+oo_option lookup plugin for openshift-ansible
+
+Usage:
+
+    - debug:
+      msg: "{{ lookup('oo_option', '<key> default=<default_value>') }}"
+
+This returns, by order of priority:
+
+* if it exists, the `opt_<key>` ansible variable. This variable is set by `bin/cluster --option <key>=<value> …`
+* if it exists, the envirnoment variable named `<key>`
+* if present, the `<default_value>` passed in the lookup statement
+* if none of the above conditions are met, empty string is returned
+'''
+
+from ansible import utils, errors
+import os
+
+# Reason: disable too-few-public-methods because the `run` method is the only
+#     one required by the Ansible API
+# Status: permanently disabled
+# pylint: disable=too-few-public-methods
+class LookupModule(object):
+    ''' oo_option lookup plugin main class '''
+
+    # Reason: disable unused-argument because Ansible is calling us with many
+    #     parameters we are not interested in.
+    #     The lookup plugins of Ansible have this kwargs “catch-all” parameter
+    #     which is not used
+    # Status: permanently disabled unless Ansible API evolves
+    # pylint: disable=unused-argument
+    def __init__(self, basedir=None, **kwargs):
+        ''' Constructor '''
+        self.basedir = basedir
+
+    # Reason: disable unused-argument because Ansible is calling us with many
+    #     parameters we are not interested in.
+    #     The lookup plugins of Ansible have this kwargs “catch-all” parameter
+    #     which is not used
+    # Status: permanently disabled unless Ansible API evolves
+    # pylint: disable=unused-argument
+    def run(self, terms, inject=None, **kwargs):
+        ''' Main execution path '''
+
+        terms = utils.listify_lookup_plugin_terms(terms, self.basedir, inject)
+
+        ret = []
+
+        for term in terms:
+            params = term.split()
+            option_name = params[0]
+
+            paramvals = {
+                'default': ''
```

Comment:
I made a try to have the lookup plugin return `None` if the value cannot be found.

It doesn’t work.
With
```python
            paramvals = {
                'default': None
```
in `oo_option.py`, the following Ansible task:
```yaml
  - debug: msg="Var toto={{ lookup('oo_option', 'toto') | default('true default') }}"
```
fails with:
```
TASK: [debug msg="Var toto={{ lookup('oo_option', 'toto') | default('true default') }}"] *** 
fatal: [localhost] => Failed to template msg="Var toto={{ lookup('oo_option', 'toto') | default('true default') }}": an unexpected type error occurred. Error was sequence item 0: expected string, NoneType found
```

I looked at how the “standard” lookup plugins are written and it seems they return an empty string:

* https://github.com/ansible/ansible/blob/devel/v1/ansible/runner/lookup_plugins/env.py#L40
* https://github.com/ansible/ansible/blob/devel/v1/ansible/runner/lookup_plugins/redis_kv.py#L67

When the lookup fails, it is `''` which is returned.